python/pyarrow/array.pxi

```diff
@@ -1259,6 +1259,54 @@ cdef class LargeListArray(Array):
         return pyarrow_wrap_array(arr.values())
 
 
+cdef class MapArray(Array):
+    """
+    Concrete class for Arrow arrays of a map data type.
+    """
+
+    @staticmethod
+    def from_arrays(offsets, keys, items, MemoryPool pool=None):
+        """
+        Construct MapArray from arrays of int32 offsets and key, item arrays
+
+        Parameters
+        ----------
+        offsets : Array (int32 type)
+        keys : Array (any type)
+        items : Array (any type)
```

Comment:
It's more relaxed than that, though -- these can be any sequences / array-like per the implementation